s3.go

```diff
@@ -67,6 +72,9 @@ func NewS3Datastore(conf Config) (*S3Bucket, error) {
 		&credentials.EnvProvider{},
 		&credentials.SharedCredentialsProvider{},
 		&ec2rolecreds.EC2RoleProvider{Client: ec2metadata.New(sess)},
+		endpointcreds.NewProviderClient(*d.Config, d.Handlers, conf.CredentialsEndpoint,
+			func(p *endpointcreds.Provider) { p.ExpiryWindow = 5 * time.Minute },
```

Comment:
This needs to be configurable based on the expiration period, right?

Reply:
endpointcreds work with both static and dynamic credentials. The dynamic ones will have an expiration field. `ExpiryWindow` permits endpointcreds to trigger a refresh for new creds at the earliest `ExpiryWindow` before the expiration deadline: https://github.com/aws/aws-sdk-go/blob/master/aws/credentials/endpointcreds/provider.go#L59-L68 

I haven't seen any default AWS expirations smaller than at least a handful of hours, but I guess I could make it clearer that this could be modified if needed, and with a somewhat smaller default value.